hack/verify-e2e-test-ownership.sh

```diff
@@ -0,0 +1,234 @@
+summary_json="${tmpdir}/summary.json"
+failures_json="${tmpdir}/failures.json"
+
+# rebuild dependencies if necessary
+function ensure_dependencies() {
+  local -r ginkgo="${KUBE_ROOT}/_output/bin/ginkgo"
+  local -r e2e_test="${KUBE_ROOT}/_output/bin/e2e.test"
+  if ! { [ -f "${ginkgo}" ] && [[ "${REUSE_BUILD_OUTPUT}" =~ ^[yY]$ ]]; }; then
+    make ginkgo
+  fi
+  if ! { [ -f "${e2e_test}" ] && [[ "${REUSE_BUILD_OUTPUT}" =~ ^[yY]$ ]]; }; then
+    hack/make-rules/build.sh test/e2e/e2e.test
+  fi
+  if ! { [ -f "${spec_summaries}" ] && [[ "${REUSE_BUILD_OUTPUT}" =~ ^[yY]$ ]]; }; then
+    "${ginkgo}" --dryRun=true "${e2e_test}" -- --spec-dump "${spec_summaries}" > /dev/null
+  fi
+}
+
+# evaluate ginkgo spec summaries against e2e test ownership polices
+# output to ${results_json}
+function generate_results_json() {
+  readonly results_jq=${tmpdir}/results.jq
+  cat >"${results_jq}" <<EOS
+  [.[] | . as { ComponentTexts: \$text, ComponentCodeLocations: \$code } | {
+      calls: [ \$text | range(1;length) as \$i | {
+        sig: ((\$text[\$i] | match("\\\[(sig-[^\\\]]+)\\\]") | .captures[0].string) // "unknown"),
+        text: \$text[\$i],
+        # unused, but if we ever wanted to have policies based on other tags...
+        # tags: \$text[\$i] | [match("(\\\[[^\\\]]+\\\])"; "g").string],
+        line: \$code[\$i] | "\(.FileName):\(.LineNumber)",
+      } + (\$code[\$i] | .FullStackTrace | match("^(.*)\\\.(.+)\\\(.*\\n") | .captures | {
+          package: .[0].string,
+          func: .[1].string
+      })],
+    } | {
+      owner: .calls[0].sig,
+      testname: .calls | map(.text) | join(" "),
+      calls,
+      policies: [(
+        .calls[0] |
+          {
+            fail: (.sig == "unknown"),
+            level: "FAIL",
+            category: "unowned_test",
+            reason: "must start with [sig-foo]",
+            found: .,
+          }, {
+            fail: (.func != "SIGDescribe"),
+            level: "FAIL",
+            category: "no_sig_describe",
+            reason: "must use top-level SIGDescribe",
+            found: .,
+          }
+        ), (
+        .calls[1:] |
+          (map(select(.sig != "unknown")) // [] | {
+            fail: . | any,
+            level: "WARN",
+            category: "too_many_sigs",
+            reason: "should not have multiple [sig-foo] tags",
+            found: .,
+          }),
+          (map(select(.func == "SIGDescribe")) // [] | {
+            fail: . | any,
+            level: "FAIL",
+            category: "nested_sig_describe",
+            reason: "must not use nested SIGDescribe",
+            found: .,
+          })
+        ), (
+        .calls |
+          (map(select(.text | match("\\\[k8s\\\.io\\\]"))) // [] | {
+            fail: . | any,
+            level: "FAIL",
+            category: "has_k8s_io",
+            reason: "should not have [k8s.io] in test names",
+            found: .,
+          }),
+          (map(select(.func == "KubeDescribe")) // [] | {
```

Comment:
BTW - given we removed `KubeDescribe` from the codebase, I think we now longer need to track that even (someone would have to reintroduce KubeDescribe again, which is very unlikely).

Reply:
Probably same for "k8s.io" above.

Reply:
Agreed. I'll remove before removing WIP label